internal/api/external.go

```diff
@@ -44,6 +45,8 @@ func (a *API) ExternalProviderRedirect(w http.ResponseWriter, r *http.Request) e
 	query := r.URL.Query()
 	providerType := query.Get("provider")
 	scopes := query.Get("scopes")
+	codeChallenge := query.Get("code_challenge")
```

Comment:
should we only grab the `codeChallenge` from the query param if the `flowType` is `pkce`?